### DISCUSSION TITLE
Explicitly specifying `index` in route for `index.js` returns a 404

### DISCUSSION THREAD

**Discussion by hgezim** [vercel/next.js#27105] [Ideas]
### What version of Next.js are you using?

10.2.3

### What version of Node.js are you using?

stackblitz

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

Stackblitz

### Describe the Bug

Create a Next.js project with this dir structure: 

/pages
  /api
  /blah
    index.js
  _app.js
  index.js

Accessing `<URL>/blah` loads `/pages/blah/index.js`.
Accessing `<URL>/blah/index` shows a 404 page. 

### Expected Behavior

Accessing `<URL>/blah/index` should load `/pages/blah/index.js`

This was the behavior in Next.js 9.x and it's a regression.

### To Reproduce

Create a Nest.js project with this structure:

```
/pages
  /blah
    index.js
  _app.js
  index.js
```

index.js can be this:

```
export default function Index() {
  return 'blah index';
}
```

Then start the project and try to access `http://localhost:3000/blah/index`.

**Comment by timneutkens** [comment]
This is not a bug, by default `/index` is not made available as `index.js` is normalized to the folder name. If you want to expose `/index` as a route you can add `/blah/index/index.js` which makes it explicitly available.
